packages/backend/src/database/migrations/20240515151028_add_custom_dimension_to_selected_dimensions.ts

```diff
@@ -0,0 +1,51 @@
+import { DBFieldTypes } from '@lightdash/common';
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+    const customBinDimensions = await knex(
+        'saved_queries_version_custom_dimensions',
+    ).select('saved_queries_version_id', 'id');
+    const customSqlDimensions = await knex(
+        'saved_queries_version_custom_sql_dimensions',
+    ).select('saved_queries_version_id', 'id');
+
+    const insertPromises = [...customBinDimensions, ...customSqlDimensions].map(
+        (customDimension) =>
+            knex('saved_queries_version_fields').insert({
+                saved_queries_version_id:
+                    customDimension.saved_queries_version_id,
+                name: customDimension.id,
+                field_type: DBFieldTypes.DIMENSION,
+                order: 99,
+            }),
+    );
+    console.debug(
+        `Selecting ${insertPromises.length} custom dimensions in saved charts`,
+    );
+    await Promise.all(insertPromises);
```

Comment:
We can refactor this to be 1 query that inserts multiple entries. This should avoid memory errors.

```
const newEntries = [
...customBinDimensions,
...customSqlDimensions
].map((customDimension) => ({
   saved_queries_version_id: customDimension.saved_queries_version_id,
   name: customDimension.id,
   field_type: DBFieldTypes.DIMENSION,
   order: customDimension.order,
}))


knex('saved_queries_version_fields').insert(newEntries);
```